### PR TITLE
Adding Subaccounts Resource

### PR DESCRIPTION
**Pull request by gmcolon**
Adding Subaccount resource

### CODE REVIEW

File: sparkpost/base.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def uri(self):
 
     def request(self, method, uri, **kwargs):
         headers = {
-            'User-Agent': 'python-sparkpost/' + sparkpost.__version__,
+            'User-Agent': 'python_sparkpost/' + sparkpost.__version__,
```
Comment (gmcolon) [inline sparkpost/base.py]: I think we need to update the module name, switch the dash to an underscore in order to import to the web module? @joelgardner @mikemelton @wynbennett 

This is not the place, just wanted to note it somewhere

**Comment by gmcolon** [comment]
Will update the version once @mikemelton 's PR is in

File: sparkpost/base.py (hunk 1 above)
Comment (mikemelton) [reply sparkpost/base.py]: The module name is just `sparkpost`. Here's how I have been testing in the console:

```
from sparkpost import SparkPost
client = SparkPost('REDACTED')
client.sending_domains.blahblahblah()
```

File: sparkpost/subaccounts.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        :param str id: the id of the subaccount to retrieve
+
+        :returns: the requested subaccount if found
+        :raises: :exc:`SparkPostAPIException` if subaccount is not found
+        """
+
+        uri = "%s/%s" % (self.uri, id)
+        return self.request('GET', uri)
+
+    def get_all(self):
```
Comment (mikemelton) [inline sparkpost/subaccounts.py]: By convention, this library calls this method `list` (see the base class).

**Review by mikemelton — COMMENTED** [review]
LGTM - just need to update that method name from `get_all` to `list`.

**Review by mikemelton — APPROVED** [review]
Love it!

**Pull request merged**